haystack/schema.py

```diff
@@ -194,8 +231,11 @@ def __repr__(self):
     def __str__(self):
         return str(self.to_dict())
 
+
+# Can we switch to an object that rather consists of List[Label] or is that nasty in the handling?
 class MultiLabel:
```

Comment:
Verify if there are other use cases outside of evaluation. If not, we should not treat it as a "primitive" as it's rather syntactic sugar to simplify the eval workflow. In that case let's  rather refactor it later.